sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/TaskKiller.java

```diff
@@ -24,36 +24,43 @@
 public final class TaskKiller {
     private static final Logger LOGGER = LoggerFactory.getLogger(TaskKiller.class);
 
-    private static final Duration killInterval = Duration.ofSeconds(5);
-    private static final Set<TaskID> tasksToKill = new HashSet<>();
-    private static final Object lock = new Object();
+    private static final Duration KILL_INTERVAL = Duration.ofSeconds(5);
+    private static final Set<TaskID> TASKS_TO_KILL = new HashSet<>();
+    private static final Object LOCK = new Object();
 
+    /**
+     * After the first task kill, this executor will be created and will periodically reissue kill invocations for any
+     * tasks which haven't produced a dead or unknown status. We do this because the Mesos kill command is best-effort.
+     */
     private static ScheduledExecutorService executor;
-    private static TaskKiller taskKiller = new TaskKiller();
+
+    /**
+     * Whether the above executor should be running. Only disabled for tests.
+     */
+    private static boolean executorEnabled = true;
 
     private TaskKiller() {
-        startScheduling();
+        // Do not instantiate
     }
 
+    /**
+     * Resets the {@link TaskKiller}'s internal state for tests.
+     *
+     * @param executorEnabled whether the background kill executor should be enabled, should only be disabled in tests
+     */
     @VisibleForTesting
-    static void shutdownScheduling() throws InterruptedException {
-        executor.shutdownNow();
-        executor.awaitTermination(killInterval.toMillis(), TimeUnit.MILLISECONDS);
-    }
+    public static void reset(boolean executorEnabled) throws InterruptedException {
```

Comment:
I assume we have to test the TaskKiller outside of its own package?

Reply:
Some tests will invoke task kill (PodResource, service tests) and have expectations on how often driver.kill() is issued, so they'd want to disable the background thread as well.